sandy/fy.py

```diff
@@ -357,6 +357,115 @@ def apply_qmatrix(self, zam, energy, decay_data):
         new_data.loc[mask, 'FY'] = fy_calc.values
         return self.__class__(new_data)
 
+    def gls_cov_update(self, zam, e, decay_data, Vy=None, kind='cumulative',
+                       threshold=None, index_columns=None):
+        """
+        IFY covariance matrix performed by GLS for a given energy and zam.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `sandy.DecayData`
+            Radioactive nuclide data for several isotopes.
+        Vy : 2D iterable, optional
+            Extra Covariance matrix (MXM). The default is None, so the method
+            uses the variance of the CFY for a given zam and e.
+        kind : `str`, optional
+            Keyword for obtaining prior sensitivity. The
+            default is 'cumulative'.
+        threshold : `int`, optional
+            Optional argument to avoid numerical fluctuations or
+            values so small that they do not have to be taken into
+            account. The default is None.
+        index_columns : `str`, optional
+            Keyword to determine if the sensitivity is the transposed one.
+            The default is None.
+
+        Returns
+        -------
+        `panda.DataFrame`
+            IFY covariance matrix performed by GLS for a given energy and zam.
+
+        """
+        S = _gls_setup(decay_data, kind, index_columns=None)
+        # Filter FY data:
+        fy_data = self.filter_by('ZAM', zam)\
+                      .filter_by("E", e).data\
+                      .set_index('ZAP')[['MT', 'DFY']]
+        # Divide the data type:
+        fy_type = fy_data.groupby('MT')
+        DIFY, DCFY = [fy_type.get_group(x)['DFY'] for x in fy_type.groups]
+        # Create the Cov object:
+        DIFY = sandy.CategoryCov.from_var(DIFY)
+        if Vy is None:
+            Vy = sandy.CategoryCov.from_var(DCFY).data
+        # Fix the index:
+        S = S.loc[DIFY.index, DIFY.index].fillna(0)
```

Comment:
this is not True, once again, sensitivities might have different indices (you will see it with chain yields, where indices are not nuclides but mass numbers)

Reply:
I put two optional index-related arguments in _gls_setup. I put them in this way since what we get from decay data is a DataFrame and we might want to reduce the size of the S. 
In the specific case of CFY and IFY, qmatrix >(CFY and IFY), therefore, by using these two arguments, we reduce the size of qmatrix by selecting only the nucleus that are related to the IFY we are analysing.

Reply:
I understand it for the first dimension of the sensitivity matrix, but not for the second. If you work with chain fission yields you don't have the index of `DIFY`. Then the second index (the one with `??`) will not work,

`S.loc[DIFY.index, ??DIFY.index??]`